src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataAnnotationsClientModelValidatorOfTAttribute.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// An implementation of <see cref="IClientModelValidator"/> which understands data annotation attributes.
+    /// </summary>
+    /// <typeparam name="TAttribute">An attribute which inherits from <see cref="ValidationAttribute"/>.</typeparam>
```

Comment:
- `The type of the attribute.`
- don't think mentioning the type restriction is worthwhile.  but if you feel strongly about it, `... Must be a <see cref="ValidationAttribute"/> subclass.`